plugins/receiver/nativelog/receiver.go

```diff
@@ -0,0 +1,72 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package nativelog
+
+import (
+	"fmt"
+	"reflect"
+
+	"google.golang.org/grpc"
+
+	logging "skywalking/network/logging/v3"
+
+	"github.com/apache/skywalking-satellite/internal/pkg/config"
+	"github.com/apache/skywalking-satellite/internal/pkg/plugin"
+	"github.com/apache/skywalking-satellite/protocol/gen-codes/satellite/protocol"
+)
+
+type Receiver struct {
+	config.CommonFields
```

Comment:
trace\metrics\log's grpc receiver looks like almost the same with each other? why not use the same grpc protocol/receiver/report_service to unify the model

Reply:
the protocol is not the same on the different scopes. https://github.com/apache/skywalking-data-collect-protocol

Reply:
@EvanLjp I think @surlymo 's idea is, more settings could be moved into a new parent, such as a thing called GRPCReceiver, to reduce the duplicated fields like `server`, `MaxBufferSpaces`,

Reply:
there is only one config named MaxBufferSpaces in grpclog-receiver. The config is for buffer some data before enqueueing. I think different receivers may have different priorities, so the receiver can have different configurations. The other configurations, such that the server, is a reference.

I think @surlymo 's idea is to use one collect service to work in multi-data scopes.

Reply:
> I think @surlymo 's idea is to use one collect service to work in multi-data scopes.But this may not fit the plug-in concept very well

This is not working. True.

____
I want to be back a little about my idea. 
```go
type Receiver struct {
	config.CommonFields
	MaxBufferSpaces int `mapstructure:"max_buffer_spaces"` // The max buffer events.

	server        *grpc.Server
	service       *LogReportService    // The gRPC request handler for logData.
	outputChannel chan *protocol.Event // The channel is to bridge the LogReportService and the Gatherer to delivery the data.

}
```
I mean is, this struct is in `nativelog`, but actually, all receiver plugin will have similar configuration besides the bind service, right? I can't see how does this related to priority.

Reply:
I would polish it now